### PR TITLE
Compute module copy edit

### PR DESCRIPTION
**Pull request by nathan-contino**
* Improved section flow and organisation
* Rewrote significant portions of the section to follow the documentation style guide and use documentation phrasing best practices (active voice, introductions to lists and figures, tables when possible, etc)
* Added an IO Board compatibility table
* Fixed a minor CSS bug where whitepaper widgets would get distorted when the description text didn't take up at least one full line.
* Added an introduction containing a description of what a Compute Module is, a description of what IO Boards are, and descriptions and (reasonably sized) images of all of our CM and IO Board models.
* Resized all images to 60% because that's plenty to see what we're displaying here.
* Added syntax highlighting to all snippets
* Improved section titles when possible
* Moved datasheets and schematics to the bottom of this section, since they're reference material
* Learned a lot about Device Tree and overlays. We should probably add a disclaimer that this section only applies to pre-CM4/CM4S devices, but I can only do so much at once :)

### CODE REVIEW

**Comment by nathan-contino** [comment]
Existing old (broken looking) whitepaper admonition:

Screenshot 2024-06-04 at 17 28 10


Updated (fixed) whitepaper admonition:

Screenshot 2024-06-04 at 17 28 31

File: documentation/asciidoc/computers/compute-module/cm-bootloader.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+==== Modify the bootloader configuration
+
+To modify the CM4 bootloader configuration:
+
+* Navigate to the `usbboot/recovery` directory
+* Replace `pieeprom.original.bin` if a specific bootloader release is required
+* Edit the default `boot.conf` bootloader configuration file; typically, at least the BOOT_ORDER must be updated:
+ ** For network boot, use `BOOT_ORDER=0xf2`
+ ** For SD/EMMC boot, use `BOOT_ORDER=0xf1`
+ ** For USB boot failing over to EMMC, use `BOOT_ORDER=0xf15`
```
Comment (lurch) [inline documentation/asciidoc/computers/compute-module/cm-bootloader.adoc:27]: The "Flash storage devices other than SD cards" section just above talks about flashing NVMe devices, so does this BOOT_ORDER section also need to mention NVMe?

File: documentation/asciidoc/computers/compute-module/cm-bootloader.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* Run `./update-pieeprom.sh` to update the EEPROM image `pieeprom.bin` image file
+* If EEPROM write-protection is required, edit `config.txt` and add `eeprom_write_protect=1`. Hardware write-protection must be enabled via software and then locked by pulling the `EEPROM_nWP` pin low
+* Run `../rpiboot -d .` to update the bootloader using the updated EEPROM image `pieeprom.bin`
+
+The `pieeprom.bin` file is now ready to be flashed to Compute Module 4.
```
Comment (lurch) [inline documentation/asciidoc/computers/compute-module/cm-bootloader.adoc:32]: "...now ready to be flashed..." - didn't that just get done by the `../rpiboot -d .` step?

File: documentation/asciidoc/computers/compute-module/cm-bootloader.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+==== Flash the bootloader EEPROM - Compute Module 4
+
+To flash the bootloader EEPROM, follow the same hardware setup as for flashing the EMMC, but also ensure `EEPROM_nWP` is NOT pulled low. Once complete, `EEPROM_nWP` may be pulled low again. Run the following command to write `recovery/pieeprom.bin` to the bootloader EEPROM:
+
+[source,console]
+----
+$ ./rpiboot -d recovery
```
Comment (lurch) [inline documentation/asciidoc/computers/compute-module/cm-bootloader.adoc:40]: Perhaps I'm just tired, but it seems a bit confusing seeing this `./rpiboot -d recovery` instruction mentioned so soon after the `../rpiboot -d .` instruction? :confused:

File: documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc [hunk 4]
```diff
@@ -1,144 +1,155 @@
-== Flashing the Compute Module eMMC
+[[flash-compute-module-emmc]]
+== Flash an image to a Compute Module
 
-[.whitepaper, title="Using the Compute Module provisioner", subtitle="", link=https://pip.raspberrypi.com/categories/685-whitepapers-app-notes/documents/RP-003468-WP/Using-the-Compute-Module-Provisioner.pdf]
+The Compute Module has an on-board eMMC device connected to the primary SD card interface. This guide explains how to flash (write) an operating system image to the eMMC storage of a single Compute Module.
```
Comment (lurch) [inline documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc:4]: Might be worth adding a comment about CM Lite here?

File: documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc [hunk 5]
```diff
@@ -1,144 +1,155 @@
...
+. Connect the IO Board to power.
 
-[,bash]
+. Then, run the `rpiboot` executable with the following command:
++
+[source,console]
 ----
-sudo dd if=raw_os_image_of_your_choice.img of=/dev/sdX bs=4MiB
+$ sudo ./rpiboot
 ----
```
Comment (lurch) [inline documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc:88]: The old version of the instructions suggested running `rpiboot` first and then powering up he IO board, but in the new instructions you seem to have switched that round? (for every OS)

File: documentation/asciidoc/computers/compute-module/cm-bootloader.adoc (hunk 1 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/compute-module/cm-bootloader.adoc:27]: I actually left most of this section alone 😆 but I'm glad you brought it up. Sprucing this file up now.

File: documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc (hunk 5 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc:88]: `rpiboot` requires the board to be powered to recognise the board as a mass storage device, AFAIK. See https://www.jeffgeerling.com/blog/2020/flashing-raspberry-pi-compute-module-on-macos-usbboot for third-party proof. Technically they could power the board _before_ they install the `usbboot` utility and dependencies, but I just wanted to reinforce that the board must be powered before you run `rpiboot`.

**Pull request merged**